cli/core/test/e2e/plugin_lifecycle/plugin_lifecycle_test.go

```diff
@@ -0,0 +1,67 @@
+// plugin_e2e_tests provides plugin command specific E2E test cases
+package plugin_e2e_tests
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-framework/cli/core/test/e2e/framework"
+)
+
+var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Plugin-lifecycle]", func() {
+	var (
+		tf        *framework.Framework
+		pluginOps framework.PluginOps
+	)
+	BeforeEach(func() {
+		tf = framework.NewFramework()
+		pluginOps = framework.NewLocalOCIPluginOps(framework.NewLocalOCIRegistry(framework.DefaultRegistryName, framework.DefaultRegistryPort))
+	})
+	Context("OCI repository based plugin lifecycle operations tests", func() {
+		When("Plugins metadata has given to generate and publish plugins", func() {
+			It("should publish plugins to repository successfully without errors", func() {
+
+				err := tf.ConfigInit()
+				Expect(err).To(BeNil())
+
+				var plugins [1]*framework.PluginMeta
+				pm := framework.NewPluginMeta()
+				pm.SetName("dummy").SetTarget("k8s").SetVersion("1.0").SetDescription("its dummy plugin").SetSHA("345").SetGroup("admin").SetArch("amd64").SetOS("darwin").SetDiscoveryType("oci").SetHidden(false).SetOptional(false)
+				plugins[0] = pm
+
+				pluginOps.GenerateScriptBasedPluginBinaries(plugins[:])
+				pluginOps.PublishPluginBinary(plugins[:])
+				pluginOps.GeneratePluginBundle(plugins[:])
+				discoveryUrls, errs := pluginOps.PublishPluginBundle(plugins[:])
+
+				//Add plugin discovery sources
+				n := len(discoveryUrls)
+				discoveryMap := make(map[string]bool)
+				for i := 0; i < n; i++ {
+					Expect(errs[i]).To(BeNil())
+					disName := "local-oci-" + framework.RandomString(3)
+					discoveryMap[disName] = true
+					do := &framework.DiscoveryOptions{
+						Name:       disName, //discovery
+						SourceType: "oci",
+						URI:        discoveryUrls[i],
+					}
+					err = pluginOps.AddPluginDiscoverySource(do)
+					Expect(err).To(BeNil())
+				}
+
+				//List plugins and make sure plugins from all above added plugin sources are exists
```

Comment:
We should verify that all the plugins exist that belong to added discovery source instead of verifying all plugin sources exists.

Reply:
yes updated.